pandas/core/indexes/base.py

```diff
@@ -3219,6 +3221,12 @@ def _convert_arr_indexer(self, keyarr):
         converted_keyarr : array-like
         """
         keyarr = com.asarray_tuplesafe(keyarr)
+
+        if needs_i8_conversion(self.dtype) and lib.infer_dtype(keyarr) == "string":
+            # weak reasoning that indexer is list of strings
+            # representing datetime or period or timedelta
+            keyarr = pd_array(keyarr, self.dtype)
```

Comment:
lets unpack PandasArray here.  AFAIK we dont have any tested support for indexing with those.

Reply:
@jbrockmendel 
you meant return type must be "ndarray" right?